warehouse/transformer/internal/rules/rules.go

```diff
@@ -16,12 +16,12 @@ type Rules func(event *types.TransformerEvent) (any, error)
 
 var (
 	DefaultRules = map[string]Rules{
-		"id":                 staticRule("messageId"),
+		"id":                 messageIDFromMetadata(),
```

Comment:
Should we add a comment here explaining why we're using the `staticRule` method for some fields, while using a different method for others?